Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -19,17 +23,35 @@ RhoFunctor::operator() ( amrex::MultiFab& mf_dst, const int dcomp, const int /*i
     auto& warpx = WarpX::GetInstance();
     std::unique_ptr<amrex::MultiFab> rho;
 
+    // Deposit charge density
+    // Call this with local=true since the parallel transfers will be handled
+    // by ApplyFilterandSumBoundaryRho
+
     // Dump total rho
     if (m_species_index == -1) {
         auto& mypc = warpx.GetPartContainer();
-        rho = mypc.GetChargeDensity(m_lev);
+        rho = mypc.GetChargeDensity(m_lev, true);
     }
     // Dump rho per species
     else {
         auto& mypc = warpx.GetPartContainer().GetParticleContainer(m_species_index);
-        rho = mypc.GetChargeDensity(m_lev);
+        rho = mypc.GetChargeDensity(m_lev, true);
     }
 
+    // Call this in case filtering is turned on
+    warpx.ApplyFilterandSumBoundaryRho(m_lev, m_lev, *rho, 0, rho->nComp());
+
+#if (defined WARPX_DIM_RZ) && (defined WARPX_USE_PSATD)
+    using Idx = SpectralAvgFieldIndex;
+    if (WarpX::use_kspace_filter) {
+        auto & solver = warpx.get_spectral_solver_fp(m_lev);
+        solver.ForwardTransform(*rho, Idx::rho_new);
```

Comment:
Should there be an index `i_comp` passed as additional argument here? I don't see a default value here
https://github.com/ECP-WarpX/WarpX/blob/82cfaf9ae48f2fc9c0c6c67bd37e98a0eff1c467/Source/FieldSolver/SpectralSolver/SpectralFieldDataRZ.H#L47-L48
Same question would be for the call `solver.BackwardTransform(*rho, Idx::rho_new);` below.

I think that this could maybe explain why the RZ Galilean test seems to fail to write `rho` to output.

Reply:
@EZoni  Thanks for noticing this! I don't know what happened but i_comp should have a default value of 0 specified in the SpectralFieldDataRZ.H file. I distinctly remember adding these default values but somehow they never made it into the final version. It's interesting that this hasn't seemed to affect the field advance since the transforms of Ez, Bz, and Jz are done with that same routine without passing in a value for i_comp. I'll submit a separate PR to fix this.

Reply:
@dpgrote I realized that the reason why this did not seem to affect the field advance is probably the fact that there was indeed a default value `i_comp=0` in the interfaces of the forward and backward FFTs exposed through the spectral solver:
https://github.com/ECP-WarpX/WarpX/blob/beb080ab67af4a610ab1a3089f374857e0e317c4/Source/FieldSolver/SpectralSolver/SpectralSolverRZ.H#L39-L40
Which explains also why #1404 was not enough to fix the Galilean RZ test here. There still seems to be `nan` appearing in the values of `rho`. They seem to appear already before the call to `ApplyFilter` and it can be seen by just running a single time step of the Galilean RZ test.

Reply:
Thanks for pointing that out about the default value. It's probably good to have the default value anyway though. I've starting looking into the nan also. When running in a single process, the nan doesn't happen.